itest/assets_test.go

```diff
@@ -501,21 +501,36 @@ func assertPendingChannels(t *testing.T, node *HarnessNode,
 	require.EqualValues(t, remoteSum, pendingRemoteBalance)
 }
 
+// haveFundingAsset returns true if the given channel has the asset with the
+// given asset ID as a funding asset.
+func haveFundingAsset(assetChannel *rfqmsg.JsonAssetChannel,
+	assetID []byte) bool {
+
+	assetIDStr := hex.EncodeToString(assetID)
+	for _, fundingAsset := range assetChannel.FundingAssets {
+		if fundingAsset.AssetGenesis.AssetID == assetIDStr {
+			return true
+		}
+	}
+
+	return false
+}
+
 func assertAssetChan(t *testing.T, src, dst *HarnessNode, fundingAmount uint64,
 	mintedAsset *taprpc.Asset) {
 
-	assetID := mintedAsset.AssetGenesis.AssetId
-	assetIDStr := hex.EncodeToString(assetID)
 	err := wait.NoError(func() error {
 		a, err := getChannelCustomData(src, dst)
 		if err != nil {
 			return err
 		}
 
-		if a.AssetInfo.AssetGenesis.AssetID != assetIDStr {
-			return fmt.Errorf("expected asset ID %s, got %s",
-				assetIDStr, a.AssetInfo.AssetGenesis.AssetID)
+		assetID := mintedAsset.AssetGenesis.AssetId
+		if !haveFundingAsset(a, assetID) {
```

Comment:
For my understanding:
In the old situation there could only be one type of AssetID as the funding asset of a custom channel. But in the new situation we can have funding assets with different AssetID's and we check if _one_ of them is the assetID we expect.